application/common/components/MfaApiClient.php

```diff
@@ -118,43 +134,128 @@ public function u2fCreateRegistration(string $appId): array
         return Json::decode($response->getBody()->getContents());
     }
 
+    /**
+     * @param string $uuid
+     * @return bool
+     * @throws GuzzleException
+     */
     public function u2fDelete(string $uuid): bool
     {
         $this->callApi('u2f/' . $uuid, 'DELETE');
         return true;
     }
 
-    public function u2fValidateAuthentication(string $uuid, $signResultJson): bool
+    /**
+     * @param string $uuid
+     * @param string $signResultJson
+     * @return bool
+     * @throws GuzzleException
+     */
+    public function u2fValidateAuthentication(string $uuid, string $signResultJson): bool
     {
         $this->callApi('u2f/' . $uuid . '/auth', 'PUT', [
             'signResult' => $signResultJson,
         ]);
         return true;
     }
 
-    public function u2fValidateRegistration(string $uuid, $signResultJson): bool
+    /**
+     * @param string $uuid
+     * @param string $signResultJson
+     * @return bool
+     * @throws GuzzleException
+     */
+    public function u2fValidateRegistration(string $uuid, string $signResultJson): bool
     {
         $this->callApi('u2f/' . $uuid, 'PUT', [
             'signResult' => $signResultJson,
         ]);
         return true;
     }
 
+    /**
+     * @param array $additionalHeaders
+     * @return array
+     * @throws GuzzleException
+     */
+    public function webauthnCreateAuthentication(array $additionalHeaders): array
+    {
+        $response = $this->callApi('webauthn/login', 'POST', [], $additionalHeaders);
+        return Json::decode($response->getBody()->getContents());
```

Comment:
Should we check for a Json decode error before just returning it?

Reply:
it doesn't return an error, though it could throw an exception. do you think we should try/catch the exception and then throw an other exception with more context?

Reply:
I was just thinking of the unhelpful emails we get that complain about a json decode error instead of telling us what it was trying to decode. I don't know if that would matter here or not.

Reply:
agreed. Matt and I chatted about it too and think catching the exceptions and adding context would be good, however since we didn't have that before and we're on a tight deadline I'll leave that for a post migration task.